packages/node_modules/@node-red/nodes/core/io/21-httprequest.html

```diff
@@ -49,13 +49,25 @@
         <label for="node-input-useAuth" style="width: 70%;"><span data-i18n="httpin.basicauth"></span></label>
         <div style="margin-left: 20px" class="node-input-useAuth-row hide">
             <div class="form-row">
+                <label for="node-input-authType"><i class="fa fa-user-secret "></i> <span data-i18n="httpin.label.authType"></span></label>
+                <select type="text" id="node-input-authType" style="width:70%;">
+                    <option value="basic"  data-i18n="httpin.basic"></option>
+                    <option value="digest" data-i18n="httpin.digest"></option>
+                    <option value="bearer" data-i18n="httpin.bearer"></option>
+                </select>
+            </div>
+            <div class="form-row node-input-basic-row">
                 <label for="node-input-user"><i class="fa fa-user"></i> <span data-i18n="common.label.username"></span></label>
                 <input type="text" id="node-input-user">
             </div>
-            <div class="form-row">
+            <div class="form-row node-input-basic-row">
                 <label for="node-input-password"><i class="fa fa-lock"></i> <span data-i18n="common.label.password"></span></label>
                 <input type="password" id="node-input-password">
             </div>
+            <div class="form-row node-input-bearer-row hide">
+                <label for="node-input-bearerToken"><i class="fa fa-lock"></i> <span data-i18n="httpin.label.bearerToken"></span></label>
```

Comment:
Hi Bart - any reason not to re-use the password field as it is effectively the same - and you can't have both :-) May make some of the following logic slightly easier as you don't then need to re-merge either this or that etc.

Reply:
Hey Dave (@dceejay),
I have been thinking the same, but I introduced a new token field because I thought you guys wouldn't like me to use the _**same field for two different purposes**_.  I can indeed re-use the password field if you like.  But is it allowed to change the '_**password**_' label to '**_token_**' (as soon as bearer authentication is selected)?  Or do you always want the label to be 'password'??

Reply:
the field can be the same as it's serving effectively the same purpose - the label can be different as the word token is more appropriate in that context.

Reply:
Hey Dave (@dceejay),
I have changed and tested everything, except from the dynamically changed label.  
Do you have any advice on how that can be done easily?
```
<label for="node-input-password"><i class="fa fa-lock"></i> <span data-i18n="common.label.password"></span></label>
<input type="password" id="node-input-password">
```
So it should become ```"common.label.bearerToken"``` ...
Thanks!

Reply:
can't you use the same .show() .hide() technique as before ?
hmm - not sure bearerToken is that "common" so it probably ought to be in "httpin.label.bearerToken"

Reply:
@dceejay,
Do you mean two label tags pointing to the same input element?  Or two span tags into a single label?

Reply:
Dave,
my time is up for this evening ...

[Seems ](https://stackoverflow.com/questions/2829936/can-an-input-field-have-two-labels)that an input tag is allowed to have multiple labels.
I added a last code snippet to switch between two labels, but the layout is not what I expected:

![token_label](https://user-images.githubusercontent.com/14224149/53053570-07b29080-34a2-11e9-8814-6975595cb6ab.gif)

Reply:
instead of two labels - have one label tag with two more span tags inside - give each span the id  you want to show/hide. (they then each have the <i... icon> and <span i18n..> inside.

Reply:
Evening Dave (@dceejay),

With the two span elements it indeed looks better now:

![bearer_token](https://user-images.githubusercontent.com/14224149/53119060-e0fe6380-354e-11e9-998e-590d9dd21ba8.gif)

I think all your feedback is implemented.
Is it ok, or does the House of Commons need to vote now?

Bart